tests/eos-test-utils.c

```diff
@@ -0,0 +1,1960 @@
+                                    SS (g_file_get_path (quit_file)))),
+                        NULL);
+
+  argv = generate_strv (EOS_UPDATE_SERVER_BINARY,
+                        SS (flag ("local-port",
+                                  port_str)),
+                        SS (flag ("timeout", "0")),
+                        SS (flag ("serve-remote",
+                                  remote_name)),
+                        NULL);
+
+  bash_script_path_base = g_strdup (g_getenv ("EOS_CHECK_UPDATE_SERVER_GDB_BASH_PATH_BASE"));
+  if (bash_script_path_base != NULL)
+    {
+      g_autoptr(GRegex) regex = g_regex_new ("XXXXXX", 0, 0, error);
+      g_autofree gchar *bash_script_path = NULL;
+      g_autoptr(GFile) bash_script = NULL;
+      g_autofree gchar *delete_me_path = NULL;
+      g_autoptr(GFile) delete_me = NULL;
+
+      if (regex == NULL)
+        return FALSE;
+
+      bash_script_path = g_regex_replace_literal (regex,
+                                                  bash_script_path_base,
+                                                  -1,
+                                                  0,
+                                                  port_str,
+                                                  0,
+                                                  error);
+      if (bash_script_path == NULL)
+        return FALSE;
+
+      bash_script = g_file_new_for_path (bash_script_path);
+      if (!generate_bash_script (bash_script, argv, envp, error))
+        return FALSE;
+
+      delete_me_path = g_strdup_printf ("%s.deleteme", bash_script_path);
+      delete_me = g_file_new_for_path (delete_me_path);
+      g_printerr ("Bash script %s generated, chmod a+x it an run it, make check will continue when %s is deleted\n",
+                  bash_script_path,
+                  delete_me_path);
+
+      if (!create_file (delete_me, NULL, error))
+        return FALSE;
+
+      while (g_file_query_exists (delete_me, NULL))
+        sleep (1);
+    }
+  else if (!test_spawn_async (argv, envp, FALSE, cmd, error))
+    return FALSE;
+
+  return TRUE;
+}
+
+static gboolean
+get_branch_file_timestamp (GFile *repo,
+                           GDateTime **out_timestamp,
+                           GError **error)
+{
+  g_autoptr(GFile) file = NULL;
+  g_autoptr(GFile) signature = NULL;
+  g_autoptr(GKeyFile) keyfile = NULL;
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(GDateTime) timestamp = NULL;
+  gint64 unix_utc;
+
+  get_branch_file_paths (repo,
+                         &file,
+                         &signature);
+
+  if (!load_key_file (file,
+                      &keyfile,
+                      error))
+    return FALSE;
+
+  unix_utc = g_key_file_get_int64 (keyfile,
+                                   "main",
+                                   "UnixUTCTimestamp",
+                                   &local_error);
+  if (local_error != NULL)
```

Comment:
Well, maybe. Unless we decide to test the branch file with timestamp set to 1970.1.1. I'd prefer to keep it like that.